public/app/features/alerting/unified/components/rules/RulesFilter.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { FormEvent, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { Button, Icon, Input, Label, RadioButtonGroup, useStyles } from '@grafana/ui';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import { css, cx } from '@emotion/css';
+import { omitBy, isUndefined } from 'lodash';
+
+import { PromAlertingRuleState } from 'app/types/unified-alerting-dto';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+import DataSourceSelect from '../DataSourceSelect';
+import { rulesFiltersSlice } from '../../state/reducers';
+import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
+
+const RulesFilter = () => {
+  const dispatch = useDispatch();
+  const [queryParams, setQueryParams] = useQueryParams();
+  const { rulesFilters } = useUnifiedAlertingSelector((state) => state.filters);
+  const { dataSource, alertState, queryString } = rulesFilters;
+  const { setDataSource, setQueryString, setAlertState, clearFilters, replaceFilters } = rulesFiltersSlice.actions;
+  const styles = useStyles(getStyles);
+  const stateOptions = Object.entries(PromAlertingRuleState).map(([key, value]) => ({ label: key, value }));
+
+  useEffect(() => {
+    if (!Object.keys(rulesFilters).length && Object.keys(queryParams).length) {
+      const queryString = queryParams['queryString'] as string | undefined;
+      const alertState = queryParams['alertState'] as string | undefined;
+      const dataSource = queryParams['dataSource'] as string | undefined;
+      const filtersToSet = omitBy({ queryString, alertState, dataSource }, isUndefined);
+      dispatch(replaceFilters({ ...filtersToSet }));
+    }
+  });
+
+  const handleDataSourceChange = (dataSourceValue: SelectableValue<string>) => {
+    dispatch(setDataSource(dataSourceValue.value as string));
+    setQueryParams({ dataSource: dataSourceValue.value });
+  };
+  const handleQueryStringChange = (e: FormEvent<HTMLInputElement>) => {
```

Comment:
it might be a good idea to debounce this event handler for 600ms or so, so it would only apply change once user stops typing

Reply:
👍